ADLStream/evaluation/base_evaluator.py

```diff
@@ -167,6 +177,7 @@ def run(self, context):
             new_results, instances = self.evaluate()
             if new_results:
                 self.write_results(new_results, instances)
+                self.write_predictions()
```

Comment:
With this new approach, we will be open the file in the `update_predictions` function, which is called by the ADLStream process. This may slow down the stream speed affecting the ADLStream performance.

As a solution, we should open the file in the `start` function and close the file at the end of the process (we may create the `end` function for this).

What do you think?